geozero/src/geojson/geojson_reader.rs

```diff
@@ -207,20 +207,30 @@ pub(crate) fn process_properties<P: PropertyProcessor>(
         // Could we provide a stable property index?
         match value {
             JsonValue::String(v) => processor.property(i, key, &ColumnValue::String(v))?,
-            JsonValue::Number(v) if v.is_f64() => {
-                processor.property(i, key, &ColumnValue::Double(v.as_f64().unwrap()))?
+            JsonValue::Number(v) => {
+                if v.is_f64() {
+                    processor.property(i, key, &ColumnValue::Double(v.as_f64().unwrap()))?
+                } else if v.is_i64() {
+                    processor.property(i, key, &ColumnValue::Long(v.as_i64().unwrap()))?
+                } else if v.is_u64() {
+                    processor.property(i, key, &ColumnValue::ULong(v.as_u64().unwrap()))?
+                } else {
+                    unreachable!()
+                }
             }
-            JsonValue::Number(v) if v.is_i64() => {
-                processor.property(i, key, &ColumnValue::Long(v.as_i64().unwrap()))?
+            JsonValue::Bool(v) => processor.property(i, key, &ColumnValue::Bool(*v))?,
+            JsonValue::Array(v) => {
+                let json_string =
+                    serde_json::to_string(v).map_err(|_err| GeozeroError::Property(key.clone()))?;
+                processor.property(i, key, &ColumnValue::Json(&json_string))?
             }
-            JsonValue::Number(v) if v.is_u64() => {
-                processor.property(i, key, &ColumnValue::ULong(v.as_u64().unwrap()))?
+            JsonValue::Object(v) => {
+                let json_string =
+                    serde_json::to_string(v).map_err(|_err| GeozeroError::Property(key.clone()))?;
+                processor.property(i, key, &ColumnValue::Json(&json_string))?
             }
-            JsonValue::Bool(v) => processor.property(i, key, &ColumnValue::Bool(*v))?,
             // For null values omit the property
             JsonValue::Null => false,
-            // Array(Vec<Value>), Object(Map<String, Value>)
-            _ => processor.property(i, key, &ColumnValue::String(&value.to_string()))?,
```

Comment:
Hmm maybe I conflated the reading and writing parts in the PR description. I don't remember exactly right now. 

I think the primary change is that we change from the `String` variant to `Json`. I don't know what was the behavior of `value.to_string()`. I assume that was not guaranteed to be the same as `serde_json::to_string`?

Reply:
I think the output is the same. I still think your change to the Reader makes sense - if we have JSON we might as well put it in a `ColumnValue::JSON` rather than a `ColumnValue::String`. I just wanted to get the changelog notes correct.

Reply:
see changelog entry at https://github.com/georust/geozero/pull/213/files#diff-f088ff93ac6d7d6d266d38c61f69ae356b4fc823196a9fd6e991f93eed1376b7R13